src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,600 @@
+import {
+  HTMLAttributes,
+  KeyboardEventHandler,
+  MouseEventHandler,
+  PointerEventHandler,
+  ReactElement,
+  forwardRef,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { GridOnScrollProps, VariableSizeGrid } from "react-window";
+import AutoSizer, { Size } from "react-virtualized-auto-sizer";
+import RowNumberColumn from "./RowNumberColumn";
+import Header from "./Header";
+import styled from "styled-components";
+import { GridContextMenuItemProps, GridProps, ItemDataType, onSelectFn } from "./types";
+import { useSelectionActions } from "./useSelectionActions";
+import { useRefCallback } from "./useRefCallback";
+import { mergeRefs } from "@/utils/mergeRefs";
+import { Cell } from "./Cell";
+import { ContextMenu } from "@/components";
+
+const NO_BUTTONS_PRESSED = 0;
+const RIGHT_BUTTON_PRESSED = 2;
+
+const GridContainer = styled.div`
+  display: flex;
+  flex-direction: column-reverse;
+  background: ${({ theme }) => theme.click.grid.body.cell.color.background.default};
+  user-select: none;
+  overflow-anchor: none;
+`;
+
+const getRenderedCursor = (children: Array<ReactElement>) =>
+  children.reduce(
+    ([minRow, maxRow, minColumn, maxColumn], { props: { columnIndex, rowIndex } }) => {
+      if (rowIndex < minRow) {
+        minRow = rowIndex;
+      }
+      if (rowIndex > maxRow) {
+        maxRow = rowIndex;
+      }
+      if (columnIndex < minColumn) {
+        minColumn = columnIndex;
+      }
+      if (columnIndex > maxColumn) {
+        maxColumn = columnIndex;
+      }
+
+      return [minRow, maxRow, minColumn, maxColumn];
+    },
+    [
+      Number.POSITIVE_INFINITY,
+      Number.NEGATIVE_INFINITY,
+      Number.POSITIVE_INFINITY,
+      Number.NEGATIVE_INFINITY,
+    ]
+  );
+
+const GridDataContainer = styled.div<{ $top: number; $left: number }>`
+  position: absolute;
+  top: 0;
+  left: 0;
+  ${({ $top, $left }) => `
+    margin-top: ${$top}px;
+    margin-left: ${$left}px;
+  `}
+`;
+
+const ContextMenuTrigger = styled.div`
+  outline: none;
+  height: 100%;
+  width: 100%;
+`;
+
+interface InnerElementTypeTypes extends HTMLAttributes<HTMLDivElement> {
+  children: Array<ReactElement>;
+}
+
+const OuterElementType = forwardRef<HTMLDivElement>((props, ref) => (
+  <div
+    ref={ref}
+    data-testid="grid-outer-element"
+    {...props}
+  />
+));
+
+export const Grid = forwardRef<VariableSizeGrid, GridProps>(
+  (
+    {
+      rowStart = 0,
+      showRowNumber = true,
+      rounded = "none",
+      showHeader = true,
+      focus,
```

Comment:
why is the focus controlled and the selection uncontrolled? shouldn't they behave in the same way?

Reply:
lets make it consistent.